[DOCS] mention deployment checklist before benchmarking (#5957)
yugabyte/yugabyte-db: docs/content/latest/benchmark/sysbench-ysql.md

The [YugabyteDB version of sysbench](https://github.com/yugabyte/sysbench) is forked from the [official](https://github.com/akopytov/sysbench) version with a few modifications to better reflect YugabyteDB's distributed nature.

{{< note title="Note" >}}

> Consult the [deployment checklist](../../deploy/checklist/) so the recommended hardware requirements are met and the
>  database is correctly configured before benchmarking. 

Reviewer: ```suggestion
To ensure the recommended hardware requirements are met and the database is correctly configured before benchmarking, review the [deployment checklist](../../deploy/checklist/)  
```
